hudi-common/src/main/java/org/apache/hudi/internal/schema/action/InternalSchemaChangeApplier.java

```diff
@@ -51,7 +51,8 @@ public InternalSchema applyAddChange(
       TableChange.ColumnPositionChange.ColumnPositionType positionType) {
     TableChanges.ColumnAddChange add = TableChanges.ColumnAddChange.get(latestSchema);
     String parentName = TableChangesHelper.getParentName(colName);
-    add.addColumns(parentName, colName, colType, doc);
+    String leafName = TableChangesHelper.getLeafName(colName);
+    add.addColumns(parentName, leafName, colType, doc);
```

Comment:
Could you add tests around nested fields to guard the changes?

Reply:
I add some UT about nested fields adding or reordering in TestTableChanges.